### ISSUE TITLE
update of the readme

### ISSUE THREAD

**Issue by jind11** [vikas95/AutoROCC#1]
hi, I am pretty interested in your paper in this repo but it seems that it has not been finished, especially for the readme file. Could you update it so that we can benefit from it? Thanks!

**Comment by vikas95** [comment]
Hi Jin,

Thanks for your interest and apologies for the delay in cleaning codes. I just updated the instructions which will help to begin the experiments and get the reported numbers.

I will clean the variable names and non-useful content by next week. Please let me know if any other detail is needed urgently.

Thanks,

**Comment by jind11** [comment]
Thanks for the quick update! I have three small questions with respect to the code:
1. For the line 10 of the file "main_MultiRC_POCC.py", since the num_sent is 5, I think the output should be "MultiRC_POCC_passage_Final_m_SENT_2_3_4_5_6_withIDF_sized_BM25_exp/" instead of "MultiRC_POCC_passage_Final_m_SENT_2_3_4_5_withIDF_sized_BM25_exp/"?
2. For the line 266 of the file "Graph_nodes.py ", why do you remove the last element of "meta_sub_graph1" by using "meta_sub_graph1[:-1]"? 
3. For the line 269 of the file "Graph_nodes.py ", the key of the "KB_terms" should the index, however, you used the "rk1" as the key to fetch the value of "KB_terms", which is not a index but a sentence. Why?
Thank you so much for your help in advance!

**Comment by vikas95** [comment]
Hi Jin,

Thanks for asking these clarifications, I have responded in the same sequence. 

1] Yes, you are right, it should be "_2_3_4_5_6_withIDF_sized_BM25_exp" because num_sent = 5. I cross-checked it with the previously saved files in both "_2_3_4_5_6_withIDF_sized_BM25_exp" and "_2_3_4_5_withIDF_sized_BM25_exp" folders.

2] "meta_subgraphs" has all the possible combinations of size 2,3,4,5 and 6 sentences. The meta_subgraph1 is looping over all these combinations. So lets say meta_subgraph1 is currently having the i^th set of meta_subgraphs - lets consider this sentence set - [2,4,7,3]. 
We want to compute overlap score between every possible pair, i.e. overlap between {(2,4), (2,7),(2,3), (4,7), (4,3)......} but we do not want to compute overlap between the last term i.e. (3,3). 
Therefore, we skip the last sentence in the current justification set for calculating overlap values. 

3] The dictionary "KB_terms" has sentence ID's as the key and non-stop words in the sentence as the value. Therefore, we use rk1 and rk2 as indexes (in line 269 and 275). 
From the example set - [2,4,7,3], rk1 and rk2 will have 2,4,7,3 as the values and we want to retrieve the KB_terms[2], KB_terms[4], KB_terms[7] and KB_terms[3].


I apologize for the missing comments, I will add detailed comments to each function once I am updating these codes. 

Thanks,

**Comment by jind11** [comment]
hi, for the second question, i agree that in line 266 of the file "Graph_nodes.py ", the last element should be removed, however, in the line 272, I think ":-1" is not correct. it should be "meta_sub_graph1[ik1 + 1:]" rather than "meta_sub_graph1[ik1 + 1:-1]". How do you think? Thanks!

**Comment by vikas95** [comment]
Yes, I think you are right here too. 
Let me double check by generating all combinations of 2 sentences (i.e. M C 2) instead of using the 2 for loops (i.e. replace the for loops in line 266 and 272). 

When I make your suggested change - "meta_sub_graph1[ik1 + 1:]" and run "BM25_vs_POCC_quality_score.py", the justification selection performance of AutoROCC improves by 1.5%. 

I'll update what happens if we replace for loops with M C 2.
Thanks!

**Comment by jind11** [comment]
I think currently the 2 for loops is the speed bottleneck for execution of this code, which is quite slow in my side for long document. Could you let me know how to replace it with a faster way? Thanks!
Another suggestion of change for this code:
In the line 275, I think this step should only be executed when ik1 equals to 0, so one line of "if ik1 == 0:" should be added before it. Although the final result would not be affected, this change could save some computation. How do you think?

**Comment by jind11** [comment]
Finally, for the line 290, there is "1+" before the variable "avg_score", however, in the equation 1 in the paper, R is not added by 1. Could you explain why you have a small change here? Thank you so much!

**Comment by vikas95** [comment]
Yes, I agree with changes in line 275, it is redundant after the first iteration. Thanks for this suggestion..

Yes, in the paper we do not add 1 to the R but we had got suggestions from audience and one of the reviewer that why not add epsilon (=1) to all the components rather than adding it to just Coverage and Overlap. 
The difference in justification selection performance is ~0.5% when we remove +1 from R VS when we add it to the R. 

I'll highlight this small change in the readme file of this repo so that other users are aware. 
Thanks!

**Comment by jind11** [comment]
Thank you so much!
Two final questions: 
1. in line 274 of the file "Graph_nodes.py ", the QA terms are also involved, which is different from what is reported in the paper, but it seems that this version is a little better than removing the QA terms. Could you give a little bit of explanations? 
2. More importantly, this code is very slow when the num_sent is set high, especially for long document, could you provide some ways to accelerate it? Thanks!

**Comment by jind11** [comment]
Sorry for bothering! I have another question:
In the line 67 of the file "main_MultiRC_POCC.py", I think we should use the function "get_POC_subgraph" rather than "get_POC_sized_BM25_subgraph". How do you think? (It is a matter of the sentence order in the gold justifications). Thanks!
